src/modules/twap/hooks/useTwapOrdersExecutions.ts

```diff
@@ -0,0 +1,55 @@
+import { useAtomValue } from 'jotai'
+import { useMemo } from 'react'
+
+import { Order, OrderInfoApi } from 'legacy/state/orders/actions'
+import { useOrders } from 'legacy/state/orders/hooks'
+
+import { useWalletInfo } from 'modules/wallet'
+
+import { DEFAULT_TWAP_EXECUTION_INFO } from '../const'
+import { twapPartOrdersAtom } from '../state/twapPartOrdersAtom'
+import { TwapOrderExecutionInfo } from '../types'
+
+export type TwapOrdersExecutionMap = { [id: string]: TwapOrderExecutionInfo }
+
+export function useTwapOrdersExecutions(ids: string[]): TwapOrdersExecutionMap {
+  const { chainId } = useWalletInfo()
+  const twapPartOrders = useAtomValue(twapPartOrdersAtom)
+
+  const allOrders = useOrders({ chainId })
+
+  const childrenIdSets = useMemo(() => {
+    return Object.keys(twapPartOrders).reduce<{ [id: string]: Set<string> }>((acc, val) => {
+      acc[val] = new Set<string>((twapPartOrders[val] || []).map((item) => item.uid))
+      return acc
+    }, {})
+  }, [twapPartOrders])
+
+  return useMemo(() => {
+    return ids.reduce<TwapOrdersExecutionMap>((acc, id) => {
+      const childrenIds = childrenIdSets[id]
+
+      if (childrenIds?.size > 0) {
+        const children = allOrders.filter((order) => childrenIds.has(order.id))
```

Comment:
Looks much better, thanks!

I'm still a bit worried about this.
Can we at least filter `allOrders` somehow to match only TWAP part orders?

Users can have lots of orders (see Elena :D) and for every twap part order we are walking down the whole list, every time `allOrders` change

Reply:
Great suggestion!
Added more optimizations